tensorflow_datasets/core/lazy_imports_lib_test.py

```diff
@@ -58,6 +58,48 @@ def test_bad_import(self):
     with self.assertRaisesWithPredicateMatch(ImportError, "extras_require"):
       _ = tfds.core.lazy_imports.test_foo
 
+  # pylint: disable=import-outside-toplevel, unused-import
+  def test_lazy_import_context_manager(self):
+    with tfds.core.try_import():
+      import pandas
+
+    self.assertTrue(hasattr(pandas, "read_csv"))
+
+  def test_import_without_context_manager(self):
+    import nltk
+    self.assertTrue(hasattr(nltk, 'tokenize'))
+
+    get_deps = "tensorflow_datasets.core.lazy_imports_lib._ALLOWED_LAZY_DEPS"
+    with mock.patch(get_deps, ["valid_module"]):
+      with self.assertRaisesWithPredicateMatch(ImportError, "valid_module"):
+        import valid_module
+
+  def test_lazy_import_context_manager_errors(self):
+    with self.assertRaisesWithPredicateMatch(ImportError, "_ALLOWED_LAZY_DEPS"):
+      with tfds.core.try_import():
+        import fake_module
+
+    get_deps = "tensorflow_datasets.core.lazy_imports_lib._ALLOWED_LAZY_DEPS"
+    with mock.patch(get_deps, ["new_module"]):
+      with tfds.core.try_import():
+        import new_module
+
+      with self.assertRaisesWithPredicateMatch(ImportError, "extras_require"):
+        new_module.some_function()
+
+  def test_nested_imports(self):
```

Comment:
This test is good, so we should keep it. However this doesn't test nested import. Importing `matplotlib` do not try to import `matplotlib.pyplot`.

We should add a test to make sure that nested work correctly:
```
with self.assertRaisesRegex(ImportError, 'y not found'):
  with tfds.core.maybe_imports():
    import x
```
With:
* ALLOWED_DEPS is x, y
* x is installed, but not y
* x.py:
```
# x.py
import y
# y should raise ImportError, instead of being a dummy module
```
To test this, you'll have to create a dummy `x.py` file.

Basically this test that if `x` is installed but `import x` fails, then adding `with tfds.core.maybe_imports()` should fail too.

Reply:
Done.
I was not sure where to create the `dummy_module_x.py`, so created it in the same directory.